test/mexpr/nestedpatterns.mc

```diff
@@ -0,0 +1,25 @@
+mexpr
+
+let classify = lam x.
+  match x with (true, true) then "one" else
+  match x with (true, false) then "two" else
+  match x with (false, true) then "three" else
+  match x with (false, false) then "four"
+  else "five" in
```

Comment:
Your interpretation of binary numbers is flawed ;)

Reply:
:)